### ISSUE TITLE
Avoid filesystem operations on the client

### ISSUE THREAD

**Issue by jcrist** [dask/dask#5380]
In certain configurations, users may have workers that have access to a filesystem that they don't have access to on their client (e.g. hdfs). It would be nice if we could move all graph-build-time filesystem operations (e.g. globbing files, etc...) to be run as tasks on the workers instead of locally. If needed this could be done by a configuration parameter, but I suspect we may want to do this always. This could also be used to parallelize filesystem operations during setup as well.

**Comment by jcrist** [comment]
Yeah, my plan here was to just wrap all setup operations in a single task and compute it for now. I think this would have to be done on a per-IO-function basis, but we may be able to abstract this later.

**Comment by mrocklin** [comment]
+1

**Comment by martindurant** [comment]
I mentioned elsewhere fsspec's dask-file-system, and the idea here is very similar, although the use case is different. Probably there is not much new that would need to be implemented.

**Comment by martindurant** [comment]
As noted elsewhere, doing FS operations via a worker is already possible - by which I mean listing files and streaming bytes. Actually parsing those bytes and/or creating graphs on workers is not possible. I would suggest either closing this issue or being more specific about what we'd like to see.

**Comment by TomAugspurger** [comment]
I think the Jim's original motivation was specifically that

> t would be nice if we could move all graph-build-time filesystem operations (e.g. globbing files, etc...) to be run as tasks on the workers instead of locally.

Unless you're saying that's already possible?

**Comment by martindurant** [comment]
Yes, you can avoid filesystem operations on the client right now, but the bytes are still streamed to the client, where any parsing and graph building are done. This means no improvement in IO parallelism, it purely solves the "workers see different files to me" situation (because of auth, etc).
